pkg/networkservice/common/connect/server.go

```diff
@@ -65,22 +76,41 @@ func NewServer(
 }
 
 func (s *connectServer) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*networkservice.Connection, error) {
-	conn, err := s.client(ctx, request.GetConnection()).Request(ctx, request.Clone())
+	clientURL := clienturlctx.ClientURL(ctx)
+
+	clientInfo := s.client(ctx, request.GetConnection())
+	conn, err := clientInfo.client.Request(ctx, request.Clone())
 	if err != nil {
+		if _, ok := s.connInfos.Load(request.GetConnection().GetId()); !ok {
+			s.closeClient(clientInfo, clientURL.String())
```

Comment:
What if there is no `clientURL` in the context?
We will come here with `injecterror.NewClient(err)` and then panic on `clientURL.String()` instead of returning an error.
Probably check from line 132 should be right after line 79.

Reply:
Done